smart_open/s3.py

```diff
@@ -330,11 +381,11 @@ def _read_from_body(self, size=-1):
 
     def read(self, size=-1):
         """Read from the continuous connection with the remote peer."""
-        if self._position >= self._content_length:
-            return b''
         if self._body is None:
-            # When the first read() after __init__() or seek(), self._body is not exist.
+            # In the first read() after __init__(), self._body does not exist.
```

Comment:
Well, technically it does exist: it is explicitly defined as `None`.

```suggestion
            # This is necessary for the very first read() after __init__().
```